src/NuGet.Core/NuGet.ProjectModel/JsonUtility.cs

```diff
@@ -51,11 +51,18 @@ internal static PackageDependency ReadPackageDependency(string property, JToken
                 versionStr == null ? null : VersionRange.Parse(versionStr));
         }
 
+        internal static JProperty WritePackageDependencyWithLegacyString(PackageDependency item)
+        {
+            return new JProperty(
+                item.Id,
+                WriteString(item.VersionRange?.ToNonSnapshotRange().ToLegacyShortString()));
```

Comment:
@nkolev92 I added `ToNonSnapshotRange()` to writting in the assets file to avoid changing the behavior in that path.

Reply:
That makes sense to me!